client/my-sites/plans-features-main/hooks/data-store/use-pricing-meta-for-grid-plans.ts

```diff
@@ -18,28 +22,88 @@ import type { IAppState } from 'calypso/state/types';
 interface Props {
 	planSlugs: PlanSlug[];
 	withoutProRatedCredits?: boolean;
+	storageAddOns?: ( AddOnMeta | null )[] | null;
+	currencyCode?: string | null;
+}
+
+/***
+ * The 2023 Pricing Grid handles prices differently than add-ons, so for the time being we need to
+ * do some conversion to calculate totals correctly.
+ *
+ * The 2023 pricing grid currently handles prices in some currencies in their smallest unit, like
+ * the Japanese Yen. For others, however, like USD, prices are not handled in their smallest unit,
+ * like the US dollar.
+ *
+ * For example, 2000 yen would be represented by the integer 2000 since a yen is the smallest unit.
+ * $20.00, however, would be represented as the integer 20, and not 2000. Since storage add-on
+ * prices are always handled in the smallest unit, we need to convert them, when appropriate, to the
+ * correct integer.
+ *
+ * We should expose a more friendly function from the format-currency package
+ */
+function convertPriceForSmallestUnit( price: number, currencyCode: string | null ) {
+	const currencyObject = getCurrencyObject( price, currencyCode || '', {
+		isSmallestUnit: true,
+	} );
+	return parseInt( currencyObject.integer.replace( /,/g, '' ) );
```

Comment:
Also strictly speaking this is not correct, since not all currencies/languages use comma as decimal separator. e.g. https://docs.oracle.com/cd/E19455-01/806-0169/overview-9/index.html . I recall there was an localization issue submitted somewhere calling out that we are not adhering these local practices, so we'd better not rely on it :)

Reply:
Good call 👍